pandas/io/html.py

```diff
@@ -821,7 +875,13 @@ def _data_to_frame(**kwargs):
     # fill out elements of body that are "ragged"
     _expand_elements(body)
     with TextParser(body, header=header, **kwargs) as tp:
-        return tp.read()
+        df = tp.read()
+
+        # Cast MultiIndex header to an Index of tuples.
+        # This maintains consistency of selection (e.g. df.columns.str[1])
+        if isinstance(df.columns, MultiIndex):
+            df.columns = Index(df.columns)
+        return df
```

Comment:
does this have any other user facing change outside of using the extract links option.

Reply:
I realised that this breaks some of the other HTML tests. I'm not sure if they are critical, but obviously a more ideal solution would work with both scenarios.